apps/researcher/src/lib/user/actions.ts

```diff
@@ -0,0 +1,24 @@
+import {clerkClient} from '@clerk/nextjs';
+import {createPersistentIri} from '@colonial-collections/iris';
+import {unstable_noStore as noStore} from 'next/cache';
+
+export async function getOrSetIri(userId: string) {
+  noStore();
+  const user = await clerkClient.users.getUser(userId);
+  if (!user) {
+    throw new Error(`User with ID ${userId} does not exist`);
+  }
+
+  let iri = user.unsafeMetadata?.iri as string | undefined;
+  if (!iri) {
+    iri = createPersistentIri();
+
+    await clerkClient.users.updateUserMetadata(userId, {
```

Comment:
To be sure: do we need to handle an error from Clerk (e.g. could `updateUserMetadata()` return null or undefined if the update fails)? Or will Clerk simply throw an error if the update is unsuccessful?